### DISCUSSION TITLE
API Version

### DISCUSSION THREAD

**Discussion by kevdogg** [Tecnativa/docker-socket-proxy#110] [General]
Hey this is more a comment than a questions.

It's pretty clear in your instructions the compatible API versions. 

I'm currently running latest version docker:

```
Client:
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.21.3
 Git commit:        afdd53b4e3
 Built:             Sun Oct 29 15:42:02 2023
 OS/Arch:           linux/amd64
 Context:           default

Server:
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.21.1
  Git commit:       1a7969545d
  Built:            Sat Sep 30 15:48:58 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.7.6
  GitCommit:        091922f03c2762540fd057fba91260237ff86acb.m
 runc:
  Version:          1.1.10
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

It was really unclear to my what API version had to do when running this container -- particularly when interacting with the watchtower container. 

Within the watchtower docker compose file I had to set the docker API version to a compatible version with environmental variable:

```
      - DOCKER_API_VERSION=1.37
```

I only noticed this when examining the docker-socket-proxy logs and started seeing log files that had API versions appended to URL:

```
192.168.32.7:33538 [08/Dec/2023:19:15:20.972] dockerfrontend dockerbackend/dockersocket 0/0/0/0/0 200 2756 - - ---- 1/1/0/0/0 0/0 "GET /v1.37/images/sha256:9807740ea1ff6522e8d61ee199243d524a9f39acdfe4d309a3a9176222ded850/json HTTP/1.1"
```

I only wish this was better explained as I'm not so sure how generalizable this solution would be to other containers.
Thanks.

**Comment by urbaman** [comment]
Hi,

I have the same problem. So, setting the DOCKER_API_VERSION=1.37 env variable makes the trick?

Thank you.
